scripts/communityScripts/decentralizedGoTo/decentralizedGoTo.js

```diff
@@ -0,0 +1,149 @@
+        messageData = JSON.parse(event);
+        if (messageData.action == "requestAddressList") {
+            goToAddresses = Settings.getValue("goToDecentral", "");
+            for (var i = 0; i < goToAddresses.length; i++) {
+
+                try {
+                    Script.require(goToAddresses[i] + "?" + Date.now());
+                }
+                catch (e) {
+                    goToAddresses.remove(goToAddresses[i]);
+                }
+            }
+            var children = goToAddresses
+                .map(function (url) { return Script.require(url + '?' + Date.now()); })
+                .reduce(function (result, someChildren) { return result.concat(someChildren); }, []);
+
+            for (var z = 0; z < children.length; z++) {
+                delete children[z].age;
+                delete children[z].id;
+                children[z]["Domain Name"] = children[z]["Domain Name"].replace(/</g, '&lt;').replace(/>/g, '&lt;');
+                children[z].Owner = children[z].Owner.replace(/</g, '&lt;').replace(/>/g, '&lt;');
+            }
+
+            permission = Entities.canRez()
+
+            var readyEvent = {
+                "action": "addressList",
+                "myAddress": children,
+                "permission": permission
+            };
+
+            tablet.emitScriptEvent(JSON.stringify(readyEvent));
+
+        } else if (messageData.action == "goToUrl") {
+            Window.location = messageData.visit;
+        } else if (messageData.action == "addLocation") {
+
+            var messageDataDomainInformation = {
+                owner: messageData.owner,
+                domainName: messageData.domainName,
+                port: messageData.Port
+            };
+
+            locationboxID = Entities.addEntity({
+                position: Vec3.sum(MyAvatar.position, Quat.getFront(MyAvatar.orientation)),
+                userData: JSON.stringify(messageDataDomainInformation),
+                serverScripts: messageData.script,
+                color: { red: 255, green: 0, blue: 0 },
+                type: "Box"
+            });
+        } else if (messageData.action == "retrievePortInformation") {
+            var readyEvent = {
+                "action": "retrievePortInformationResponse",
+                "goToAddresses": goToAddresses
+            };
+
+            tablet.emitScriptEvent(JSON.stringify(readyEvent));
+            
+        }
+    }
+
+    function onMenuItemEvent(menuItem) {
+        if (menuItem == "Subscribe to new GoTo provider") {
+            goToAddress = Settings.getValue("goToDecentral", "");
+            var arrayLength = goToAddress.length;
+            var prom = Window.prompt("Enter URL to GoTo JSON.", "");
+            if (prom) {
+                goToAddress[arrayLength] = prom;
+                Settings.setValue("goToDecentral", goToAddress);
+                Menu.addMenuItem("GoTo > Unsubscribe from GoTo provider", prom);
+            }
+        } else {
```

Comment:
This branch needs an `if(…)` guard, otherwise it is executed for any menu item event, not just ones related to this app.